### ISSUE TITLE
BRAF V600E in GRCh38 instance

### ISSUE THREAD

**Issue by zhx828** [genome-nexus/genome-nexus#436]
The following page returns annotation for BRAF V600E in GRCh38
http://grch38.genomenexus.org/variant/7:g.140753336A%3ET

But I don't understand why GN shows ENST00000496384 as transcript id while  ENST00000288602 is the actual canonical transcript? http://grch38.genomenexus.org/ensembl/canonical-transcript/hgnc/BRAF

**Comment by leexgh** [comment]
@zhx828 Good catch, there might be some bugs, somehow GN marks `ENST00000496384` with `"canonical": "1"` in the annotation. @inodb @onursumer Any thoughts?
https://grch38.genomenexus.org/annotation/7%3Ag.140753336A%3ET?fields=annotation_summary
```
    {
      "exon": "15/19",
      "transcript_id": "ENST00000496384",
      "hgvsp": "ENSP00000419060.2:p.Val600Glu",
      "hgvsc": "ENST00000496384.7:c.1799T>A",
      "variant_allele": "T",
      "codons": "gTg/gAg",
      "protein_id": "ENSP00000419060",
      "protein_start": 600,
      "protein_end": 600,
      "gene_symbol": "BRAF",
      "gene_id": "ENSG00000157764",
      "amino_acids": "V/E",
      "hgnc_id": "HGNC:1097",
      "canonical": "1",
      "polyphen_score": 0.927,
      "polyphen_prediction": "probably_damaging",
      "sift_score": 0.0,
      "sift_prediction": "deleterious",
      "refseq_transcript_ids": [
        "NM_001354609.2"
      ],
      "consequence_terms": [
        "missense_variant"
      ]
    },
```

**Comment by inodb** [comment]
@leexgh the "canonical": "1" comes from Ensembl and is not necessarily the canonical transcript that we have assigned for that hugo symbol. It's best to use `annotation_summary.transcriptConsequenceSummary` to get the canonical transcript

**Comment by leexgh** [comment]
@inodb I see, yeah it's weird in the `annotation_summary.transcriptConsequenceSummary` it's also `ENST00000496384`.  Actually GN frontend is using `annotation_summary.transcriptConsequenceSummary` for most of header info pills, so that's why we show ENST00000496384 as transcript id
```
    "transcriptConsequenceSummary": {
      "transcriptId": "ENST00000496384",
      "codonChange": "gTg/gAg",
      "aminoAcids": "V/E",
      "aminoAcidRef": "V",
      "aminoAcidAlt": "E",
      "entrezGeneId": "673",
      "consequenceTerms": "missense_variant",
      "hugoGeneSymbol": "BRAF",
      "hgvspShort": "p.V600E",
      "hgvsp": "p.Val600Glu",
      "hgvsc": "ENST00000496384.7:c.1799T>A",
      "proteinPosition": {
        "start": 600,
        "end": 600
      },
      "refSeq": "NM_001354609.2",
      "variantClassification": "Missense_Mutation",
      "exon": "15/19"
    }
```

**Comment by inodb** [comment]
@leexgh srry for the confusion. It has to do with the API not correctly setting the default `isoformOverrideSource` when missing (should be `uniprot`). Try e.g.:

https://grch38.genomenexus.org/annotation/7%3Ag.140753336A%3ET?fields=annotation_summary&isoformOverrideSource=uniprot

vs

https://grch38.genomenexus.org/annotation/7%3Ag.140753336A%3ET?fields=annotation_summary

**Comment by inodb** [comment]
This btw is not that great in the API design imo. It would've prolly been better to have separate fields for each canonical transcript (uniprot/mskcc/maybe others) instead of the field changing based on a filter parameter. The filter parameter should filter the results, not change fields in the result. Mostly historical reasons that is this way, but something to consider for API response 2.0 :)
